astropy/cosmology/core.py

```diff
@@ -216,7 +216,7 @@ class FLRW(Cosmology):
         density at z=0.  If this is set to None (the default), any
         computation that requires its value will raise an exception.
 
-    name : str, optional
+    name : str or None, optional
```

Comment:
`None` really just is a placeholder to make this optional, no? Not sure this adds to the clarity...

Reply:
But technically speaking that's an allowed value. And it's a common way of having a placeholder so most people will recognize that. If you call this function form some other function, passing through user arguments, you might want to know that `None` is an allowed value.

Reply:
OK, fair point! I still feel it is slightly better without, but this PR is an improvement regardless.

Reply:
While numpydoc has no hard and fast rule on describing allowed values for parameters, it does say "For the parameter types, be as precise as possible." When I check with `mypy` it flags these as ``typing.Optional[str]``

Reply:
That `typing.Optional[str]` is indeed how I would interpret `name` as well - if it is given, it should be `str` (even if `None` is also OK). But again, we're at the hair-splitting level here!

Reply:
Agreed that it's hair splitting. Were it ups to me, I would include None, especially because we are a long way away from adding type annotations.

Reply:
> None seems there just as a placeholder, not something you'd hope people actually pass in.

Agreed, except that users abstracting over a cosmology would use the ``cls._init_signature.bind_partial()`` and that fills in a `None`, which we want to note is a valid input (it should be because it's the default value, but I've seen and been party to some truly heinous things like default values that can raise an error depending on a different value)